packages/api-observable/src/Base.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright 2017-2018 @polkadot/ui-observable authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { RxApiInterface, RxApiInterface$Method } from '@polkadot/api-rx/types';
+import { Method } from '@polkadot/jsonrpc/types';
+
+import { Observable, combineLatest } from 'rxjs';
+import { defaultIfEmpty, map } from 'rxjs/operators';
+import { StorageFunction } from '@polkadot/api-codec/StorageKey';
+import assert from '@polkadot/util/assert';
+import isUndefined from '@polkadot/util/is/undefined';
+
+type MapFn<R, T> = (combined: R) => T;
+
+const defaultMapFn = (result: any): any =>
+  result;
+
+export default class ApiBase {
```

Comment:
Split out the implementation from the UI -

- Base - just the basics, i.e. a nice wrapper for raw calls and storage
- Calls - Anything that maps to an actual call on the API, i.e. getBlock
- Queries - Anything that just simply wraps a storage query
- Combined - Multiple observables combined into a single result